.github/workflows/build_and_test.yml

```diff
@@ -283,6 +284,179 @@ jobs:
         name: unit-tests-log-${{ matrix.modules }}-${{ matrix.comment }}-${{ matrix.java }}-${{ matrix.hadoop }}-${{ matrix.hive }}
         path: "**/target/unit-tests.log"
 
+  # Maven Test: build Spark and run the tests for specified modules using maven.
+  maven-test:
```

Comment:
Hm. should we make a separate file? I think at least we can leverage `uses`

Reply:
Sorry, didn't understand what you meant.

Do you mean we shoud merge `maven-test` into an existing `build`?

Reply:
Already move to a new file, but I don't know how to trigger a test for `build_and_test_maven.yml` in the current pr now ... :(

Reply:
Sorry, `On pull request update / Notify test workflow ` always failed, I close this one and open a new one : https://github.com/apache/spark/pull/42200

Reply:
@Yikun  do you know how to check a new workflow?

Reply:
IIUC, the workflow will be shown in personal actions page:
https://github.com/LuciferYang/spark/actions/workflows/build_main.yml

Reply:
I accidentally deleted this branch, the new PR is https://github.com/apache/spark/pull/42203